api/agent/drivers/docker/docker.go

```diff
@@ -102,13 +146,34 @@ func NewDocker(conf drivers.Config) *DockerDriver {
 		}
 	}
 
+	go NewImageCleaner(driver, context.Background())
+
+	logrus.Infof("Max cache size: %v\n", conf.MaxImageCacheSize)
+	driver.imageCache = NewCache(int64(conf.MaxImageCacheSize))
+
+	imagesBeforeLoad, err := driver.docker.ListImages(context.Background())
+
 	if conf.DockerLoadFile != "" {
 		err = loadDockerImages(driver, conf.DockerLoadFile)
 		if err != nil {
 			logrus.WithError(err).Fatalf("cannot load docker images in %s", conf.DockerLoadFile)
 		}
 	}
 
+	go func() {
+		images, err := driver.docker.ListImages(context.Background())
```

Comment:
probably don't need to double ListImages(). It's safer to get the list of images from loadDockerImages().

Reply:
It looked like loadDockerImages does not return a list of what it loaded.